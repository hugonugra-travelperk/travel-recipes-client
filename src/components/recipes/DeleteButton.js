import {useNavigate} from "react-router-dom";
import {Button} from '../../components/base/Button'
import useAxios from '../../hooks/useAxios'

export default function DeleteButton(props) {
  let navigate = useNavigate();

  const url = '/recipes/' + props.recipe.id + '/'  

  const { dispatch, response, loading, error } = useAxios({
    method: 'DELETE',
    url,
  }, performAction)

  function performAction(res, err) {
    if(err === null && res === '') {
      props.handleUpdate();
      navigate('/', {replace: true});
    } else {
      window.alert(err.message);
    }
  }

  return (
    <Button primary onClick={
      () => { 
        if (window.confirm('Are you sure you wish to delete this item?')) dispatch()
      }
    }> Delete </Button>
  );
}
