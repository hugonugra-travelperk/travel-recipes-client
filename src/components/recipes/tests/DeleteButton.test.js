import {render, cleanup} from "@testing-library/react"
import "@testing-library/jest-dom";
import { BrowserRouter } from "react-router-dom";
import DeleteButton from "../DeleteButton"

afterEach(cleanup);

it("renders", () => {
  const {asFragment} = render(
    <BrowserRouter>
      <DeleteButton handleUpdate={()=>{}} recipe = {{'id':'1', 'name': 'test text'}}/>
    </BrowserRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
