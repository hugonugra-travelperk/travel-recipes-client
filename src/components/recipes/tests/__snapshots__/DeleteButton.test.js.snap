// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <button
    class="sc-bdvvtL lcNcVF"
  >
     Delete 
  </button>
</DocumentFragment>
`;
